MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+        let bridge_transfer_id = bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        atomic_bridge_store::add(bridge_transfer_id, details);  
+    
+        // Burn the amount from the initiator  
+        atomic_bridge::burn(initiator_address, amount);  
+    
+        // Emit an event with nonce  
+        event::emit(  
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }  
+
+    /// Completes a bridge transfer by the initiator.  
+    ///  
+    /// @param caller The signer representing the bridge operator.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.  
+    
+    /// @abort If the caller is not the bridge operator.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) {  
+        atomic_bridge_configuration::assert_is_caller_operator(caller);  
+        let ethereum_address = ethereum::ethereum_address(initiator);
+        let details = atomic_bridge_store::create_details(  
+            ethereum_address,  
+            recipient,  
+            amount,  
+            nonce  
+        );
+
+        // asserts that parameters are correct  
+        assert!(bridge_transfer_id == bridge_transfer_id(&details), 0x1);
+        atomic_bridge_store::add(bridge_transfer_id, details);
+        
+        // Mint to recipient  
+        atomic_bridge::mint(recipient, amount);  
+
+        event::emit(  
+            BridgeTransferCompletedEvent {  
+                bridge_transfer_id,  
+                initiator,  
+                recipient,  
+                amount,  
+                nonce,  
+            },  
+        );  
+    }  
+
+```
+
+## Verification
+
+1. **Correctness**:
+   - Simulate multiple transaction scenarios to ensure robustness.
+   - Test edge cases, including relayer downtime and batch processing.
+
+2. **Security Implications**:
+   - Conduct audits focused on the simplified design.
+   - Implement rate-limiting safeguards and validate parameters in contracts.
+
+3. **Performance Impacts**:
+   - Benchmark gas costs and transaction throughput.
+
+4. **Validation Procedures**:
+   - Perform an audit and thorough testing, alongside an open invite to the community to verify the bridge.
+   - Seek community feedback and incorporate suggestions.
+
```

Comment:
What happens if the relayer or network is down for a long time?
What are the trust assumptions?

From the slack discussion I understand the following:
- if the relayer is down or L2 is down, _L2_complete_tx_ will be processed later when the stopped component is up and running again. This is reasonable with the assumption of partial synchrony (eventually the network is synchronous for a period of time and all messages are delivered etc). 
- this requires that:
   - we monitor/health check the relayer, and possibly have a fault-tolerance mechanism in place 
   -  we maintain a _reliable mempool_  on the L2 to make sure tx relayed by the relayer are not lost (e.g. if we reboot the L2 network).
   
It looks like the trust base should  include a reliable  mempool on L2.
When bridging from L2 to L1,  we have to make sure we send our Tx  to an _honest client_ or otherwise it could be censored, delayed etc.

Overall, the only way for a user to get some funds back is to bridge back from L2 to L1.
The duration of a refund is then the time it takes to safely bridge back (minutes, hours, days?)

Reply:
A refund would then be minutes. But I would advocate that we cannot operate it for them, they must process it themselves.

Reply:
> we maintain a reliable mempool on the L2 to make sure tx relayed by the relayer are not lost (e.g. if we reboot the L2 network)

we may be ok without a reliable mempool initially. However, the relayer must check after some time that submitted `complete` transactions have been successful, or otherwise retry them

Reply:
Yes, makes sense.